samples/matter/lock/CMakeLists.txt

```diff
@@ -16,6 +16,8 @@ if(CONF_FILE)
     get_filename_component(CONF_FILE_NAME ${CONF_FILE} NAME CACHE)
 endif()
 
+set(EXTRA_DTC_OVERLAY_FILE ${CMAKE_CURRENT_LIST_DIR}/../common/src/dts/${BOARD}_retention.overlay)
```

Comment:
It's not a good idea to set it unconditionally. I assume we cannot use Kconfig here, because it's before loading Zephyr env, so maybe the only option is to do that manually, by adding overlay to the build command.

Reply:
It is done only within the https://github.com/ArekBalysNordic/sdk-nrf/pull/13/commits/d72f68ac69a5aab84be6655ee6355b88add19ecf commit, so only for test event purpose.